pkg/deploy/deployer.go

```diff
@@ -146,3 +151,63 @@ func (d *deployer) getParameters(ps map[string]interface{}) *arm.Parameters {
 
 	return parameters
 }
+
+func (d *deployer) deploy(ctx context.Context, t map[string]interface{}, p *arm.Parameters, rgName, deploymentName, vmssName string) (err error) {
+	for i := 0; i < 3; i++ {
+		d.log.Printf("deploying %s", deploymentName)
+		err = d.deployments.CreateOrUpdateAndWait(ctx, rgName, deploymentName, mgmtfeatures.Deployment{
+			Properties: &mgmtfeatures.DeploymentProperties{
+				Template:   t,
+				Mode:       mgmtfeatures.Incremental,
+				Parameters: p.Parameters,
+			},
+		})
+		if serviceErr, ok := err.(*azure.ServiceError); ok &&
+			serviceErr.Code == "DeploymentFailed" &&
+			i < 1 {
+			// on new RP deployments, we get a spurious DeploymentFailed error
+			// from the Microsoft.Insights/metricAlerts resources indicating
+			// that rp-lb can't be found, even though it exists and the
+			// resources correctly have a dependsOn stanza referring to it.
+			// Retry once.
+			d.log.Print(err)
+			continue
```

Comment:
Don't we need to call `removeFailedScaleset` if we get `DeploymentFailed`?

Reply:
I don't think so, not if it's a spurious error as the comment states.

Reply:
If it wasn't a spurious error, the VMSS should get removed on the next loop.